### PR TITLE
Fix #18733: Crash on delete all instruments and undo

### LINKED ISSUE #18733: Crash when delete all instruments and Undo (one score)
### Issue type

Crash or freeze

### Bug description

_No response_

### Steps to reproduce

1. Open [Smoke Test 11.zip](https://github.com/musescore/MuseScore/files/12111574/Smoke.Test.11.zip)
2. In Instruments panel, Select all instruments and delete
3. Undo

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/90187801/a57deaea-0fec-43cd-bb77-162a48668658

### MuseScore Version

4.1.1 nightly builds

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows10

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #18733

### CODE REVIEW

**Comment by RomanPudashkin** [comment]
@mike-spa could you please fix [this crash](https://github.com/musescore/MuseScore/issues/18733#issuecomment-1644285375) as well? You can simply check if pitchClass >=0 && pitchClass < columnBottom.size(). Or apply the patch that Casper suggested

**Comment by mike-spa** [comment]
I actually couldn't reproduce that crash, even though I am building with the address sanitizer option turned on.
image
Am I doing something wrong?

**Comment by RomanPudashkin** [comment]
Is pitchClass negative for you when you open the score from the issue? If so, the application could potentially crash when we try to use it as an index

**Comment by cbjeukendrup** [comment]
The problem is that because it's on a TAB staff, `note->line()` is `INVALID_LINE`, which equals `-10000`. This causes the expression `(line + 700) % 7` to be `-4`, which (should) cause a crash when being used as an index. 

@mike-spa The `MUE_BUILD_ASAN` option has currently no effect on Windows, so that might be the problem. Or are you using Linux?

**Comment by mike-spa** [comment]
Aaaahhhh ok, yes I work on windows, that's why I don't see that. Let me just switch to my Linux machine. I'll upload the fix asap

**Comment by mike-spa** [comment]
At a first look, not a single thing inside layoutChords3 seems to be applicable to tab staves, so I think tabs shouldn't even call this function. That's a bit dangerous to do now though, so I've just followed Casper's suggestion

**Comment by DmitryArefiev** [comment]
@mike-spa Tested on Win10. I can get a **crash** with the same score but with closed parts using these steps:
1. Open [Smoke Test 11a.zip](https://github.com/musescore/MuseScore/files/12124793/Smoke.Test.11a.zip)
2. Select all
3. Backspace to clear measures
4. In Instrument panel, Delete all instruments
5. Undo
6. Redo 
7. Undo



https://github.com/musescore/MuseScore/assets/90187801/f35f6fc7-d2ee-4b12-88ba-c1bf08f14710

Can you check please is it the same or not?

**Comment by DmitryArefiev** [comment]
@mike-spa Sorry.. I checked it for regression and it happens too in **4.1** (#18733 doesn't happen in **4.1**)

**Comment by mike-spa** [comment]
Checking

**Review by RomanPudashkin — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
@mike-spa I still have a **crash** with some additional steps (it doesn't occur in 4.1)
1. Delete instruments from Flutes to Violoncellos
2. Undo
3. Delete all instruments 
4. Undo


https://github.com/musescore/MuseScore/assets/90187801/5fec3714-7588-4bee-bc32-af790051150b

**Comment by mike-spa** [comment]
Ok, it turns out that all this instability is due to this commit 785deb16cbce21f68db0e9cfd41479cef93bd2e9. I was trying to work around it, but I think at this point I just have to revert it, which means going back and find a different solution to the problem that commit was tring to fix

**Comment by mike-spa** [comment]
Ok, as far as I can see and understand, the commit I've reverted was trying to fix a bug that also exists in 4.0, so it's not a regression and we don't need to fix it. @DmitryArefiev let me know if you find any more instability issues with this PR, otherwise I think it's good to go

**Review by RomanPudashkin — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
@mike-spa All good. Thanks!

Tested #18733 on Win10, Mac13, LinuxUbuntu 22.04 - **FIXED**

File: src/engraving/layout/v0/chordlayout.cpp [hunk 1]
```diff
@@ -2292,10 +2292,11 @@ void ChordLayout::layoutChords3(const MStyle& style, const std::vector<Chord*>&
     int prevSubtype = 0;
     int prevLine = std::numeric_limits<int>::min();
 
+    bool isTab = staff->isTabStaff(tick);
     for (int i = nNotes - 1; i >= 0; --i) {
         Note* note     = notes[i];
         Accidental* ac = note->accidental();
-        if (ac && ac->subtype() == prevSubtype && note->line() == prevLine) {
+        if (ac && (isTab || (ac->subtype() == prevSubtype && note->line() == prevLine))) { // TODO: probably this function shouldn't be called at all for tab staves
```
Comment (MarcSabatella) [inline src/engraving/layout/v0/chordlayout.cpp:2299]: I think you're probably right.  The main doubt for me is the one line to layout stemSlash a bit further down.  But it seems that happens other places too and I'm not sure there anything special that should need to be done during this pass.

**Pull request merged**